### PR TITLE
Add spdlog

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **spdlog/1.4.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by danimtb** [comment]
The create step is not able to find `fmt`:

```
CMake Error at source_subfolder/CMakeLists.txt:129 (find_package):

  By not providing "Findfmt.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "fmt", but
  CMake did not find one.

  Could not find a package configuration file provided by "fmt" with any of
  the following names:

    fmtConfig.cmake
    fmt-config.cmake

  Add the installation prefix of "fmt" to CMAKE_PREFIX_PATH or set "fmt_DIR"
  to a directory containing one of the above files.  If "fmt" provides a
  separate development package or SDK, be sure it has been installed.
```

Maybe this is because the `fmt` package does not contain *.cmake files anymore. You could try using ´cmake_find_package´ generator

**Comment by uilianries** [comment]
I've added `cmake_find_package` generator. it has been worked like charm!

File: recipes/spdlog/1.4.x/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(cmake_wrapper)
+
+message(STATUS "Conan CMake Wrapper")
```
Comment (Croydon) [inline recipes/spdlog/1.4.x/CMakeLists.txt]:
```suggestion
```

**Comment by Croydon** [comment]
We should update this to 1.4.2

File: recipes/spdlog/1.4.x/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "spdlog", "cmake"))
+
+    def package_id(self):
+        if self.options.header_only:
+            self.info.header_only()
+
+    def package_info(self):
+        if self.options.header_only:
+            self.cpp_info.defines = ["SDPLOG_HEADER_ONLY", "SPDLOG_FMT_EXTERNAL"]
```
Comment (Croydon) [inline recipes/spdlog/1.4.x/conanfile.py]:
```suggestion
            self.cpp_info.defines = ["SPDLOG_HEADER_ONLY", "SPDLOG_FMT_EXTERNAL"]
```

File: recipes/spdlog/1.4.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        else:
+            self.cpp_info.libs = tools.collect_libs(self)
+            self.cpp_info.defines = ["SDPLOG_COMPILED_LIB", "SPDLOG_FMT_EXTERNAL"]
```
Comment (Croydon) [inline recipes/spdlog/1.4.x/conanfile.py]:
```suggestion
            self.cpp_info.defines = ["SPDLOG_COMPILED_LIB", "SPDLOG_FMT_EXTERNAL"]
```

**Comment by Croydon** [comment]
Also our Bincrafters recipe got some updates in the meantime which we should port

**Comment by uilianries** [comment]
Finally working! 

Yeah!

**Pull request merged**